src/routes/safe/components/Apps/ManageApps.js

```diff
@@ -0,0 +1,172 @@
+import GnoForm from '~/components/forms/GnoForm'
+import { composeValidators, required } from '~/components/forms/validator'
+import Img from '~/components/layout/Img'
+import appsIconSvg from '~/routes/safe/components/Transactions/TxsTable/TxType/assets/appsIcon.svg'
+
+const FORM_ID = 'add-apps-form'
+
+const StyledText = styled(Text)`
+  margin-bottom: 19px;
+`
+
+const StyledTextFileAppName = styled(TextField)`
+  && {
+    width: 340px;
+  }
+`
+
+const AppInfo = styled.div`
+  margin: 36px 0 24px 0;
+
+  img {
+    margin-right: 10px;
+  }
+`
+const APP_INFO = { iconUrl: appsIconSvg, name: '', error: false }
+
+type Props = {
+  appList: Array<{
+    id: string,
+    iconUrl: string,
+    name: string,
+    disabled: boolean,
+  }>,
+  onAppAdded: (app: any) => void,
+  onAppToggle: (appId: string, enabled: boolean) => void,
+}
+
+const ManageApps = ({ appList, onAppAdded, onAppToggle }: Props) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const [appInfo, setAppInfo] = useState(APP_INFO)
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true)
+
+  const onItemToggle = (itemId: string, checked: boolean) => {
+    onAppToggle(itemId, checked)
+  }
+
+  const handleSubmit = () => {
+    setIsOpen(false)
+    onAppAdded(appInfo)
+  }
+
+  const urlValidator = (value: string) => {
+    return /(?:^|[ \t])((https?:\/\/)?(?:localhost|[\w-]+(?:\.[\w-]+)+)(:\d+)?(\/\S*)?)/gm.test(value)
+      ? undefined
+      : 'Please, provide a valid url'
+  }
+
+  const safeAppValidator = async (value) => {
+    const appInfo = await getAppInfoFromUrl(value)
+
+    if (appInfo.error) {
+      setAppInfo(APP_INFO)
+      return 'This is not a valid Safe app.'
+    }
+
+    setAppInfo({ ...appInfo })
+  }
+
+  const onFormStatusChange = ({ pristine, valid, validating }) => {
+    if (!pristine) {
+      setIsSubmitDisabled(validating || !valid || appInfo.error)
+    }
+  }
+
+  const getAddAppForm = () => {
+    return (
+      <GnoForm
+        initialValues={{
+          appUrl: '',
+          agreed: true,
+        }}
+        onSubmit={handleSubmit}
+        testId={FORM_ID}
+      >
+        {() => (
+          <>
+            <StyledText size="xl">Add custom app</StyledText>
+            <DebounceValidationField
+              component={TextField}
+              label="App URL"
+              name="appUrl"
+              placeholder="App URL"
+              type="text"
+              validate={composeValidators(required, urlValidator, safeAppValidator)}
+            />
+
+            <AppInfo>
+              <Img alt="Token image" height={55} src={appInfo.iconUrl} />
+              <StyledTextFileAppName label="App name" readOnly value={appInfo.name} />
+            </AppInfo>
+
+            <FormSpy
+              onChange={onFormStatusChange}
+              subscription={{
+                valid: true,
+                pristine: true,
+                validating: true,
+              }}
+            />
+
+            <Field
```

Comment:
This checkbox is by default ticked for me ... does this make sense?

Reply:
I'll ask to Lukas

Reply:
uncheked by default